### PR TITLE
[myq] Remove the access token when the thing is removed

### PR DESCRIPTION
**Pull request by lolodomo**
Related to #14818

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Review by digitaldan — APPROVED** [review]
LGTM

File: bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java [hunk 1]
```diff
@@ -162,10 +162,20 @@ public void dispose() {
         stopPolls();
         OAuthClientService oAuthService = this.oAuthService;
         if (oAuthService != null) {
-            oAuthService.close();
+            oAuthFactory.ungetOAuthService(getThing().toString());
```
Comment (jlaur) [inline bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java:165]: Perhaps also call `removeAccessTokenRefreshListener` here first? See:

https://github.com/openhab/openhab-addons/blob/5d65507584d6c66f457829d9e88d2be0864b3cfe/bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java#L598-L607

**Review by jlaur — CHANGES_REQUESTED** [review]
Thanks! One additional improvement suggested.

File: bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java:165]: Yes, I did not see that.

Comment (lolodomo) [reply bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java:166]: Added

**Review by jlaur — APPROVED** [review]

**Pull request merged**